src/modules/sync/lib/convert-tree-to-sync-options.ts

```diff
@@ -9,78 +9,136 @@ type PushOptionsWithSelections = {
 		plugins?: string[];
 		themes?: string[];
 		uploads?: string[];
+		contents?: string[];
 	};
 };
 
-const isSyncOption = ( value: string ): value is SyncOption => {
-	return Object.keys( SYNC_OPTIONS ).includes( value );
+const normalizePath = ( path: string ): string =>
+	path.replace( /^\/?wp-content\//, '' ).replace( /^\/+|\/+$/g, '' );
+
+const traverseSelected = (
+	nodes: TreeNode[] | undefined,
+	visit: ( node: TreeNode ) => void
+): void => {
+	if ( ! nodes?.length ) return;
+	for ( const node of nodes ) {
+		if ( node.checked ) {
+			visit( node );
+		} else if ( node.indeterminate && node.children?.length ) {
+			traverseSelected( node.children, visit );
+		}
+	}
+};
+
+const collectPathIds = ( nodes: TreeNode[] | undefined ): string[] => {
+	const out: string[] = [];
+	traverseSelected( nodes, ( node ) => {
+		if ( node.pathId ) out.push( node.pathId );
+	} );
+	return out;
+};
+
+type Categories = {
+	plugins: Set< string >;
+	themes: Set< string >;
+	uploads: Set< string >;
+	contents: Set< string >;
 };
 
-const collectPathIds = ( nodes: TreeNode[], pathIds: string[] = [] ): string[] => {
-	nodes.forEach( ( node ) => {
-		if ( node.checked && node.pathId ) {
-			pathIds.push( node.pathId );
-		} else if ( node.indeterminate && node.children ) {
-			collectPathIds( node.children, pathIds );
+const categorizeSelectedPathsInPlace = (
+	nodes: TreeNode[] | undefined,
+	categories: Categories
+): void => {
+	traverseSelected( nodes, ( node ) => {
+		if ( ! node.path ) return;
+
+		const p = normalizePath( node.path );
```

Comment:
```suggestion
		const nodePath = normalizePath( node.path );
```
Use whatever name you like, but let's use something descriptive.